src/storage/v2/property_store.hpp

```diff
@@ -19,6 +20,45 @@
 
 namespace memgraph::storage {
 
+struct SearchResult {
+  std::uint64_t id;
+  nlohmann::json document;
+  double score;
+};
+
+class ExternalStoreMock {
+ private:
+  std::string INDEX = "";  // placeholder (Tantivy supports multiple indices)
+  std::map<std::string, std::map<std::uint64_t, nlohmann::json>> storage{};  // index_name: {document_id: document}
+
+ public:
+  enum class Consistency : uint8_t { DEFAULT };
+
+  void CreateAllPropsIndex(std::string index_name, LabelId label, std::string tokenizer = "DUMMY_TOKENIZER",
+                           Consistency consistency = Consistency::DEFAULT);
+
+  void CreateIndex(std::string index_name, LabelId label, std::vector<PropertyId> properties,
+                   std::string tokenizer = "DUMMY_TOKENIZER", Consistency consistency = Consistency::DEFAULT);
+
+  void DropIndex(std::string index_name) { storage.erase(index_name); }
+
+  /// @brief Add document with given ID
+  /// @param id Id of graph element
+  /// @param document Document storing the properties
+  void AddDocument(std::uint64_t id, nlohmann::json document) { storage[INDEX][id] = document; }
```

Comment:
Let's split document into: `metadata` (e.g. txid, gid, is_deleted, is_node) and `data` (properties)

Reply:
On it

Reply:
N/A anymore